AprilSylph/XKit-Rewritten
# Panorama: Update for vertical navigation; implement custom width

### Description


This reimplements Panorama after the vertical navigation UI changes, adds a preference for the maximum post column width, and fixes a number of elements that were not correctly widened in Panorama's previous iteration (some of which I never bug reported, like link elements).

~~(The default width, pre-reblog-redesign, corresponds to the 3-column masonry view on the explore page. Post reblog redesign it's totally arbitrary at the moment.)~~

~~It is somewhat awkward due to Redpop's use of the same wrapper elements around the main page content and right sidebar regardless of the contents of said main page content. Without the use of `:has`, it is very difficult to do behavior conditional on e.g. the settings page rather than the post column being the main page contents when using a pure CSS solution. I thought a little bit about whether `pageModifications` would be helpful here, but it's not immediately obvious to me what one would target and what one would toggle. I bet that's eventually the best path forward though.~~

~~At the moment, this can't handle the reblog redesign, so it should stay a draft until that experiment ends in either direction.~~

Resolves #814, resolves #985, resolves #1026, resolves #1109, resolves #1041, resolves discussion #391, resolves discussion #853, resolves discussion #1145.

Resolves #860.

Resolves half of #937, the tag page portion; blog view panorama is more complicated.

### Technical notes

The breakpoint value constants are in the end magic numbers, but they're broken out as they appear in the Tumblr CSS that dictates them, so it should be possible to follow them/edit them if and when Tumblr makes changes.

`onStorageChanged` is used just to replace the `clean().then(main)` storage change behavior, which caused pretty spectacular visual thrashing when typing in the preference box, with just running `main()`. A bit unorthodox, I know; the semantically correct version would basically just be putting everything in `main` in a differently-named function and calling it, though, which seems a bit silly.

### Testing steps


- Confirm that Panorama widens post columns based on the viewport width until they reach the width in pixels specified by the added preference.
- Confirm that media elements in posts are, depending on the preference, either widened to the post width (and heightened proportionally) or left with their normal width and centered. This includes:
	- full width single images
	- <300px width images (should be left alone)
	- images with source links
	- images in asks
	- photosets (should be widened when images are side-by-side; individual images should max out at 540px if the expand setting is disabled)
	- polls
	- native videos
	- native videos with vanilla videos enabled
	- audio elements
	- spotify embeds
	- youtube embeds
	- link cards with images
	- various ad containers

--

- Confirm that Panorama functions with endless scrolling enabled and with endless scrolling disabled.
- Confirm that j/k scrolling works on a widened timeline.
- Confirm that resizing the viewport horizontally, affecting post height, doesn't (completely) break post rendering on a widened timeline.

--

- Confirm that Panorama functions:
	- on the dashboard (only in the widest Tumblr desktop layout), including the "tumblr tv" tab
	- on the "channel" pages (including draft, queue and blaze)
	- in the inbox
	- on communities pages
	- in explore pages if and only if a single column of posts is selected
	- in search pages if and only if a single column of posts is selected
	- on collections pages if and only if a single column of posts is selected
	- on patio columns if and only if "wide column" is selected
- Note that Panorama doesn't function:
	- in the floating blog view modal
	- in the "embedded" blog view modal from direct navigation to a blog
	- in settings/blog settings
	- in tagged pages*
	- on https://www.tumblr.com/communities*
	- on individual community posts*
	- in communities settings*
	- on the likes page**
	- on the following/followers pages
	- in the "tumblr tv" ui

*probably would be active here in an ideal world
**I honestly think this is just a redpop bug; other "postColumn" elements are `<main>`s but the likes route uses a div for some reason
